src/contract.rs

```diff
@@ -531,25 +600,37 @@ fn get_input_price(
     input_amount: Uint128,
     input_reserve: Uint128,
     output_reserve: Uint128,
+    fee_percent: Uint128,
 ) -> StdResult<Uint128> {
     if input_reserve == Uint128::zero() || output_reserve == Uint128::zero() {
         return Err(StdError::generic_err("No liquidity"));
     };
 
-    let input_amount_with_fee = input_amount
-        .checked_mul(Uint128::new(997))
-        .map_err(StdError::overflow)?;
+    let fee_reduction_percent = FEE_SCALE_FACTOR - fee_percent;
+    let input_amount_with_fee = input_amount.full_mul(fee_reduction_percent);
     let numerator = input_amount_with_fee
-        .checked_mul(output_reserve)
+        .checked_mul(Uint256::from(output_reserve))
```

Comment:
This can overflow as `2^128 * 10_000 * 2^128 > 2^256`. Unlikely to happen, but we may need to go up to `Uint512` here.

Reply:
Hmm... yeah I agree this is unlikely, but will bump up to 512 bit uints to be safe.